### PR TITLE
Navigation component

### PR DESCRIPTION
**Pull request by youssef-night**

### CODE REVIEW

File: libs/ui/components/navigation/breadcrumb/props.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface BreadcrumbElement {
```
Comment (ThomasNight) [inline libs/ui/components/navigation/breadcrumb/props.ts:1]: with interfaces add `I` prefix
```suggestion
export interface IBreadcrumbElement {
```

File: libs/ui/components/navigation/tabs/tab/index.tsx [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 import { ITabProps } from './props';
-import { Flex, Text } from '@chakra-ui/react';
+import COLORS from '../../../../constants/colors';
+import Text from '../../../contents/text';
+import FONTS from '../../../../constants/fonts';
 
 const Tab = ({ children, onSelect, active }: ITabProps): JSX.Element => {
     // Render
     return (
         <Text
```
Comment (ThomasNight) [inline libs/ui/components/navigation/tabs/tab/index.tsx:10]: use Paragraph
and Box or Flex for parent to have onClick props

**Review by ThomasNight — CHANGES_REQUESTED** [review]

File: libs/ui/components/inputs/button-icon/index.stories.tsx [hunk 3]
```diff
@@ -109,11 +109,11 @@ RemoveMemberSmall.args = {
     gap: '10px',
     backgroundColor: '#FFFFFF',
     hoverBackgroundColor: '#F2F4F9',
-    iconComponent: (isHovered) => (
+    iconComponent: ({ isHovered }) => (
         <Icon
             pointerEvents='none'
             name={'removeSmall'}
-            stroke={isHovered ? '#F46363' : '#8F95B2'}
+            stroke={{ isHovered } ? '#F46363' : '#8F95B2'}
```
Comment (ThomasNight) [inline libs/ui/components/inputs/button-icon/index.stories.tsx:116]:
```suggestion
            stroke={ isHovered  ? '#F46363' : '#8F95B2'}
```
I don't know why you put it in an object

File: libs/ui/components/inputs/button-icon/index.stories.tsx [hunk 4]
```diff
@@ -125,11 +125,11 @@ RemoveMemberXL.args = {
     gap: '10px',
     backgroundColor: '#FFFFFF',
     hoverBackgroundColor: '#F2F4F9',
-    iconComponent: (isHovered) => (
+    iconComponent: ({ isHovered }) => (
         <Icon
             pointerEvents='none'
             name={'removeLarge'}
-            stroke={isHovered ? '#F46363' : '#8F95B2'}
+            stroke={{ isHovered } ? '#F46363' : '#8F95B2'}
```
Comment (ThomasNight) [inline libs/ui/components/inputs/button-icon/index.stories.tsx:132]: same here

File: libs/ui/components/inputs/button-icon/props.ts [hunk 5]
```diff
@@ -2,11 +2,11 @@ import { IconButtonProps } from '@chakra-ui/react';
 import React from 'react'
 export default interface IButtonIconProps extends IconButtonProps {
     size: ButtonSize,
-    iconComponent: (isHovered: boolean) => React.ReactElement,
+    iconComponent: ({ isHovered }: State) => React.ReactElement,
     hoverBackgroundColor?: string,
     handleOnClick?: () => void,
 }
 
 export enum ButtonSize {
```
Comment (ThomasNight) [inline libs/ui/components/inputs/button-icon/props.ts:10]: move it to types.ts

File: libs/ui/components/inputs/button-icon/types.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+type State = {
```
Comment (ThomasNight) [inline libs/ui/components/inputs/button-icon/types.ts:1]: To generic
E.g. ButtonIconState

File: libs/ui/components/navigation/breadcrumb/props.ts (hunk 1 above)
Comment (ThomasNight) [reply libs/ui/components/navigation/breadcrumb/props.ts:1]: Todo

Comment (ThomasNight) [reply libs/ui/components/navigation/breadcrumb/props.ts:1]: And move it to types.ts

File: libs/ui/components/navigation/tabs/tab/index.tsx [hunk 7]
```diff
@@ -1,19 +1,21 @@
 import React from 'react';
 import { ITabProps } from './props';
-import { Flex, Text } from '@chakra-ui/react';
+import COLORS from '../../../../constants/colors';
+import Text from '../../../contents/text';
+import FONTS from '../../../../constants/fonts';
+import { Box } from '@chakra-ui/react';
 
 const Tab = ({ children, onSelect, active }: ITabProps): JSX.Element => {
```
Comment (ThomasNight) [inline libs/ui/components/navigation/tabs/tab/index.tsx:8]:
```suggestion
const Tab = ({ children, onSelect, isActive }: ITabProps): JSX.Element => {
```

**Review by ThomasNight — CHANGES_REQUESTED** [review]

File: libs/ui/components/inputs/button-icon/index.stories.tsx (hunk 3 above)
Comment (youssef-night) [reply libs/ui/components/inputs/button-icon/index.stories.tsx]: I just forgot to change it after a "replace all"

File: libs/ui/components/inputs/button-icon/types.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export type ButtonIconState = {
+    isHovered: boolean
+}
+
+export enum ButtonSize {
+    XS = "1.75rem", SMALL = "2rem", REGULAR = "2.25rem", LARGE = "2.5rem", XL = "2.75rem", XXL = "3rem"
```
Comment (ThomasNight) [inline libs/ui/components/inputs/button-icon/types.ts:6]: Enum should not be used to set values
XS = "XS"

and in the component if (size == XS) return "1.75rem"

**Review by ThomasNight — APPROVED** [review]

**Pull request merged**